brightwind/load/load.py

```diff
@@ -2284,6 +2296,156 @@ def apply_cleaning(data, cleaning_file_or_df, inplace=False, sensor_col_name='Se
     return data
 
 
+def _apply_cleaning_rule(df, condition_col, target_cols, comparator_value, comparison_operator_id, replacement_value, 
+                         date_from, date_to):
+    """Apply cleaning rule based on a single column to replace values on a list of columns
+
+    :param df:                      Data to be cleaned.       
+    :type df:                       pandas.DataFrame
+    :param condition_col:           Column that the cleaning rule is based on
+    :type condition_col:            str
+    :param target_cols:             Column to apply the cleaning rule to clean the data of
+    :type target_cols:              List[str]
+    :param comparator_value:        Threshold value to use in the cleaning rule, defined by 
+                                    the cleaning_rule_json format.
+    :type comparator_value:         float
+    :param comparison_operator_id:  Operator code (1-6) defined by the cleaning_rule_json format, to be used with
+                                    variable operator_dict to define the operator used on the condition column.
+                                    Code number corresponds to the following operators:
+                                        1: is less than (<),
+                                        2: is less than or equal (≤),
+                                        3: is greater than (>),
+                                        4: is greater than or equal (≥),
+                                        5: equals (=),
+                                        6: not equals (≠)
+    :type comparison_operator_id:   int
+    :param replacement_value:       Value or string to replace the data in the target_cols with
+    :type replacement_value:        str | np.nan
+    :param dates_from:              List of datetimes that the cleaning rule should be applied from for each column. 
+    :type dates_from:               str
+    :param dates_to:                List of datetimes that the cleaning rule should be applied until for each column. If
+                                    None is present the data is cleaned until the end of the file.
+    :type dates_to:                 str| None
+    :return:                        Cleaned data
+    :rtype:                         pandas.DataFrame
+    """
+
+    result_df = df
+    op = OPERATOR_DICT[comparison_operator_id]
+    mask = op(df[condition_col], comparator_value)
+    if date_to:
+        date_filter = (df.index >= date_from) & (df.index < date_to)
+    else:
+        date_filter = (df.index >= date_from)
+
+    for i, col in enumerate(target_cols):
+        mask_date_range = pd.Series(False, index=df.index)
+        mask_date_range = mask & date_filter
```

Comment:
@olivia-bentley this first assignment of mask_date_range is overwritten by the next one. So, is there any need for this line?